src/main/scala/spray/json/ProductFormats.scala

```diff
@@ -56,9 +56,11 @@ trait ProductFormats extends ProductFormatsInstances {
       try reader.read(x.fields(fieldName))
       catch {
         case e: NoSuchElementException =>
-          deserializationError("Object is missing required member '" + fieldName + "'", e)
+          deserializationError("Object is missing required member '" + fieldName + "'", e, fieldName)
+        case DeserializationException(msg, cause, None) =>
+          deserializationError(msg, cause, fieldName)
```

Comment:
This would then become:
case DeserializationException(msg, cause, fieldNames) =>
  deserializationError(msg, cause, fieldName :: fieldNames)
```scala

```